spinalcordtoolbox/deepseg/inference.py

```diff
@@ -77,3 +91,158 @@ def segment_and_average_volumes(model_paths, input_filenames, options):
+    # Copy the file to the temporary directory using shutil.copyfile
+    path_img_tmp = os.path.join(tmpdir, os.path.basename(path_img))
+    shutil.copyfile(path_img, path_img_tmp)
+    logger.info(f'Copied {path_img} to {path_img_tmp}')
+
+    # define inference patch size and center crop size
+    crop_size = (64, 192, -1)
+    inference_roi_size = (64, 192, 320)
+
+    # define the dataset and dataloader
+    test_loader, test_post_pred = ds_monai.prepare_data(path_img_tmp, tmpdir, crop_size=crop_size)
+    batch = next(iter(test_loader))
+
+    # Run nnUNet prediction
+    print('Starting inference...')
+    start = time.time()
+
+    # run inference
+    test_input = batch["image"].to(torch.device("cpu"))
+    batch["pred"] = sliding_window_inference(test_input, inference_roi_size, mode="gaussian",
+                                             sw_batch_size=4, predictor=predictor, overlap=0.5, progress=False)
+    pred = ds_monai.postprocessing(batch, test_post_pred)
+
+    end = time.time()
+    print('Inference done.')
+    total_time = end - start
+    print(f'Total inference time: {int(total_time // 60)} minute(s) {int(round(total_time % 60))} seconds')
+
+    # this takes about 0.25s on average on a CPU
+    # image saver class
+    _, fname, ext = extract_fname(path_img)
+    postfix = "seg"
+    target = f"_{postfix}"
+    pred_saver = SaveImage(
+        output_dir=tmpdir, output_postfix=postfix, output_ext=ext,
+        separate_folder=False, print_log=False)
+    # save the prediction
+    fname_out = os.path.join(tmpdir, f"{fname}_{postfix}{ext}")
+    logger.info(f"Saving results to: {fname_out}")
+    pred_saver(pred)
+
+    return [fname_out], [target]
+
+
+def segment_nnunet(path_img, tmpdir, predictor):
+    """
+    This script is used to run inference on a single subject using a nnUNetV2 model.
+
+    Author: Jan Valosek, Naga Karthik
+    Original script: https://github.com/ivadomed/model_seg_sci/blob/4184bc22ef7317b3de5f85dee28449d6f381c984/packaging/run_inference_single_subject.py
+    """
+    # Copy the file to the temporary directory using shutil.copyfile
+    path_img_tmp = os.path.join(tmpdir, os.path.basename(path_img))
+    shutil.copyfile(path_img, path_img_tmp)
+    logger.info(f'Copied {path_img} to {path_img_tmp}')
+
+    # Get the original orientation of the image, for example LPI
+    orig_orientation = get_orientation(Image(path_img_tmp))
+
+    # Get the orientation used by the model
+    if "SCI" in predictor.plans_manager.dataset_name:
+        model_orientation = "RPI"
+    else:
+        assert "levels" in predictor.plans_manager.dataset_name  # rootlets dataset
+        model_orientation = "LPI"
+
+    # Reorient the image to model orientation if not already
+    img_in = Image(path_img_tmp)
+    if orig_orientation != model_orientation:
+        logger.info(f'Changing orientation of the input to the model orientation ({model_orientation})...')
+        img_in.change_orientation(model_orientation)
```

Comment:
should we reorient too monai also?

Reply:
Reorientation in monai is done internally [here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/0c07d1973a764acef1ea80c113b54b97e39a5aba/spinalcordtoolbox/deepseg/monai.py#L181) after loading and preprocessing the input image.

Reply:
Does `monai.transform.Orientationd` (and/or `monai.transforms.SaveImage`) automatically change the output segmentation back to the original orientation of the input image? (i.e. "undoing" the transform) 

If not, then that's something that may need to be added, as I don't see us explicitly specifying the original orientation anywhere.

Reply:
Yes, the undo-ing is done by [this transform](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/b0d35b8a05fb5af6246e8f18545174281c2c3ee4/spinalcordtoolbox/deepseg/monai.py#L165-L171), which is called [here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/b0d35b8a05fb5af6246e8f18545174281c2c3ee4/spinalcordtoolbox/deepseg/monai.py#L201). So, yes, the segmentation is brought back to the native resolution of the input image!

Reply:
Ahhh, perfect! Thank you!